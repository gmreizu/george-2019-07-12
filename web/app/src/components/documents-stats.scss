.documents-stats {
    display: flex;
    justify-content: space-between;

    h2 {
        margin: 0;
        margin-bottom: 24px;
        padding: 0;
    }
}