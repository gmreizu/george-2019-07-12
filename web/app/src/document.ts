/** Document represents a document stored on the service. */
export class Document {
    public static compareByTime(a: Document, b: Document) {
        return b.time.getTime() - a.time.getTime()
    }

    constructor(
        public id: string,
        public title: string,
        public size: number,
        public path: string,
        public time: Date = new Date(),
    ) {
    }
}