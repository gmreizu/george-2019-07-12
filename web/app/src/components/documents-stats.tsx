import * as React from "react";
import { Document } from "../document";
import { formatBytes } from "../util";
import "./documents-stats.scss";

interface Props {
    readonly documents: Document[]
}

export const DocumentsStats: React.SFC<Props> = (props: Props): JSX.Element => {
    const { documents } = props

    const documentsCount = documents.length
    const totalSize = documents.reduce((sum, document) => sum + document.size, 0)

    return (
        <div className="documents-stats">
            <h2>{documentsCount} {documentsCount === 1 ? "document" : "documents"}</h2>
            <div>Total size: {formatBytes(totalSize)}</div>
        </div>
    )
}