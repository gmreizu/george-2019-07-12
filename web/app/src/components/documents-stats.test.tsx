import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { Document } from "../document";
import { DocumentsStats } from "./documents-stats";

const testDocuments = [
    new Document("1", "Doc 1", "/uploads/doc1.jpg", 123),
    new Document("2", "Doc 2", "/uploads/doc1.jpg", 234),
]

describe("DocumentStats", () => {
    it("renders with documents", () => {
        const wrapper = shallow(<DocumentsStats documents={testDocuments} />)
        expect(wrapper.contains(<div>Total size: 357 Bytes</div>)).toEqual(true)
    })

    it("renders when no documents passed", () => {
        shallow(<DocumentsStats documents={[]} />)
    })

    it("renders correctly", () => {
        const tree = renderer
            .create(<DocumentsStats documents={testDocuments} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
