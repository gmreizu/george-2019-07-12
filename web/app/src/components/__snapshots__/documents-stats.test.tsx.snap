// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentStats renders correctly 1`] = `
<div
  className="documents-stats"
>
  <h2>
    2
     
    documents
  </h2>
  <div>
    Total size: 
    357 Bytes
  </div>
</div>
`;
